### PR TITLE
simplify loading

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

**Comment by jipkim** [comment]
I've tried it, but still not working.
I think the environment variable setting has to be updated.
Also, library directories are different from where they are located.

**Comment by KristofferC** [comment]
Ah, yeah if the directory is different on mac that will not work. I'll try fix a mac VM today.

**Comment by jipkim** [comment]
1. MKL library directory will vary depending on the version.
This is the directory on my mac that automatically generated after installation.
mkl library directory: `/opt/intel/compilers_and_libraries_2017.4.181/mac/mkl/lib`
(I'm not sure this is correct library for PARDISO.jl though)

2. In case of '`libpardiso500-MACOS-X86-64.dylib`' library file, I downloaded it directly from the website. So, I don't know where to put. 
Moreover, it seems like I need to install 'gcc' to link the library (I installed `gcc 7.1.0` using Homebrew on terminal: `brew install gcc`).

3. Environment variable must be set.

**Comment by jipkim** [comment]
**Libraries that I'm using:**
**1) PARDISO 5.0:**  `libpardiso500-MACOS-X86-64.dylib` (http://www.pardiso-project.org/download)
**2) Intel Math Kernel Library for macOS:** `m_mkl_2017.3.181.dmg `(https://software.intel.com/en-us/mkl)

**Comment by KristofferC** [comment]
Ok, I got a macOS VM and managed to get MKL to pass on it.

Regarding the Pardiso library, I am having some problems with the paths (likely because I don't know anything about mac development). Could you try and see if MKL is working now?

File: deps/build.jl [hunk 1]
```diff
@@ -3,20 +3,13 @@ using Compat
...
     "libpardiso500-GNU461-X86-64",
     "libpardiso500-GNU472-X86-64",
     "libpardiso500-GNU481-X86-64",
     "libpardiso"
 ]
-end
-end
-
-@static if is_windows() begin
-    const LIBPARDISONAMES = ["libpardiso500-WIN-X86-64.dll", "libpardiso"]
```
Comment (tkelman) [inline deps/build.jl:17]: did you mean to delete this?

Comment (KristofferC) [reply deps/build.jl:17]: It was supposed to go in the vector above. Will fix. Many thanks.

**Comment by jipkim** [comment]
I've updated `src/Pardiso.jl`, `src/project_pardiso.jl`, `src/mkl_pardiso.jl`, `deps/build.jl` files as above.
And this is what I see now.

```
julia> Pkg.build("Pardiso")
INFO: Building Pardiso
found libpardiso at /Users/SingingKim/.julia/v0.6/Pardiso/deps/libpardiso500-MACOS-X86-64.dylib, using it
did not find MKLROOT key, assuming MKL is not installed
```

```
julia> using Pardiso
INFO: Recompiling stale cache file /Users/SingingKim/.julia/lib/v0.6/Pardiso.ji for module Pardiso.
WARNING: Pardiso did not manage to load, error thrown was: ErrorException("could not load library \"libgomp\"\ndlopen(libgomp.dylib, 9): image not found")
```

```
julia> PardisoSolver()
ERROR: pardiso library was not loaded
Stacktrace:
 [1] Pardiso.PardisoSolver() at /Users/SingingKim/.julia/v0.6/Pardiso/src/project_pardiso.jl:16
```

**Comment by KristofferC** [comment]
See the Readme about "Set the MKLROOT environment variable. See the MKL getting started manual for a thorough guide how to set this variable correctly."

**Comment by jipkim** [comment]
@KristofferC It's finally working! Thanks a lot!

**Pull request merged**